test/server/admin/prometheus_stats_test.cc

```diff
@@ -47,16 +46,16 @@ class PrometheusStatsFormatterTest : public testing::Test {
   void addCounter(const std::string& name, Stats::StatNameTagVector cluster_tags) {
     Stats::StatNameManagedStorage name_storage(baseName(name, cluster_tags), *symbol_table_);
     Stats::StatNameManagedStorage tag_extracted_name_storage(name, *symbol_table_);
-    counters_.push_back(alloc_.makeCounter(name_storage.statName(),
-                                           tag_extracted_name_storage.statName(), cluster_tags));
+    test_scope_ptr_->counters_.push_back(alloc_.makeCounter(
```

Comment:
I don't think we need to access the private member vars of TestScope -- we can just use the Scope API to add these, right?

Reply:
Possibly, but then we would have to change the `PrometheusStatsFormatterTest` class even more? 

My idea was to keep the test setup as close as possible the current version, that's why I ended up adding a new `TestScope` class and accessing its private vars. Do we need to change that?

Reply:
I think it would not be more of a change that what you have already. You can just call test_scope_ptr_->counterFromStatName().

I noted this in the header file you added, but I'm not clear why you needed to subclass IsolatedStore and IsolatedScope, rather than just using them directly.